app/models/mixins/scanning_operations_mixin.rb

```diff
@@ -0,0 +1,263 @@
+require 'xmldata_helper'
+require 'yaml'
+
+module ScanningOperationsMixin
+  include Vmdb::Logging
+  WS_TIMEOUT = 60
+
+  def save_metadata_op(xmlFile, type, jobid=nil)
+    begin
+      Timeout.timeout(WS_TIMEOUT) do # TODO: do we need this timeout?
+        _log.info "target [#{guid}],  job [#{jobid}] enter"
+        _log.info "target [#{guid}] found target object id [#{id}], job [#{jobid}]"
+        MiqQueue.put(
+          :target_id   => id,
+          :class_name  => self.class.base_class.name,
+          :method_name => "save_metadata",
+          :data        => Marshal.dump([xmlFile, type]),
+          :task_id     => jobid,
+          :zone        => my_zone,
+          :role        => "smartstate"
+        )
+        _log.info "target [#{guid}] data put on queue, job [#{jobid}]"
+      end
+    rescue Exception => err
+      _log.log_backtrace(err)
```

Comment:
I think that's true for code in gems/pending, but code under app is MIQ-specific, so it should be OK.